python/shaped_bloom_filter/filter.py

```diff
@@ -114,6 +96,30 @@ def serialize(self) -> bytes:
         }
         return pickle.dumps(serializable, protocol=pickle.HIGHEST_PROTOCOL)
 
+    def load(self, serialized_bloom_filter: bytes):
+        """
+        Restore a serialized bloom filter into the existing context.
+        Computationally faster than creating a new filter from scratch.
+        """
+        self.__del__()
```

Comment:
Is this needed? I feel this will just free memory for next gc pass but wont impact the object, lazy mem delete

Reply:
Yes, this is needed because we're specifically de-allocating memory in C, not in Python. And this has to happen by manually.

Reply:
But doesnt that happen when you reassign to same attributes ? 

Or, what happens if you don't do this? everytime you load mem increases and its not properly cleaned?

Reply:
Nope, it doesn't happen. This is not Python-managed memory, but C-allocated. If we don't do this, then there's gonna be memory leak till the system OOMs.

Reply:
But you are calling the python finalizer method here right? That method is the default one managed by python, did you overwrite it somewhere? 

Did you test the OOM?

Reply:
Oh what i meant is that is the default python finalizer and it wont have any effect so that's why I asked if you reimplemented it, that I see you did here https://raw.githubusercontent.com/shaped-ai/bloom/performance-improvement-change/python/shaped_bloom_filter/filter.py

Then is alright. My concern was that you were calling the default one. I would still think is good to test this

Reply:
Yep, we have our own `__del__` implementation.

And yes, we have tested this before and the memory leak is there indeed. Gave it another go now too and the memory leak is definitely there.